app/components/post_textbox/post_textbox.js

```diff
@@ -551,7 +592,8 @@ export default class PostTextbox extends PureComponent {
                         />
                         {this.renderSendButton()}
                     </View>
-                </View>
+                </View>}
+                {channelIsArchived && this.archivedView(theme, style)}
```

Comment:
@hmhealey It doesn't have to be here, it could be separate component, but I put it here as an easy way to ensure that every instance of the post textbox is replaced with the message. I'm not opposed to this being refactored into a separate component at a later date.